src/odemis/gui/util/test/img_test.py

```diff
@@ -601,6 +601,80 @@ def test_first(self):
+        numpy.testing.assert_array_equal(ovv_im_new[5:, 5:, :], numpy.zeros((6, 6, 3)))
+
+        # Test tile that goes beyond the borders of the image top left
+        ovv_im = model.DataArray(numpy.zeros((5, 5, 3)))
+        ovv_im.metadata[model.MD_PIXEL_SIZE] = (1, 1)
+        tile = model.DataArray(255 * numpy.ones((2, 2, 3)))
+        tile.metadata[model.MD_POS] = (-3, 3)  # top-left corner, one pixel diagonal to start of ovv
+        tile.metadata[model.MD_PIXEL_SIZE] = (1, 1)
+        ovv_im_new = insert_tile_to_image(tile, ovv_im)
+        self.assertEqual(ovv_im_new[0][0][0], 255)  # first pixel white (all three dims), rest black
+        numpy.testing.assert_array_equal(ovv_im.flatten()[3:],
+                                         numpy.zeros(len(ovv_im.flatten()[3:])))
+
+        # Test tile that goes beyond the borders of the image bottom right
+        ovv_im = model.DataArray(numpy.zeros((5, 5, 3)))
+        ovv_im.metadata[model.MD_PIXEL_SIZE] = (1, 1)
+        tile = model.DataArray(255 * numpy.ones((2, 2, 3)))
+        tile.metadata[model.MD_POS] = (3, -3)  # bottom-right corner, one pixel diagonal to end of ovv
+        tile.metadata[model.MD_PIXEL_SIZE] = (1, 1)
+        ovv_im_new = insert_tile_to_image(tile, ovv_im)
+        self.assertEqual(ovv_im_new[4][4][0], 255)
+        numpy.testing.assert_array_equal(ovv_im.flatten()[:-3],
+                                         numpy.zeros(len(ovv_im.flatten()[3:])))
+
+        # Test tile that lies completely outside the overview image
+        ovv_im = model.DataArray(numpy.zeros((5, 5, 3)))
+        ovv_im.metadata[model.MD_PIXEL_SIZE] = (1, 1)
+        tile = model.DataArray(255 * numpy.ones((2, 2, 3)))
+        tile.metadata[model.MD_POS] = (10, -10)
+        tile.metadata[model.MD_PIXEL_SIZE] = (1, 1)
+        ovv_im_new = insert_tile_to_image(tile, ovv_im)
+        numpy.testing.assert_array_equal(ovv_im_new, numpy.zeros((5, 5, 3)))
+
+    def test_merge(self):
+        """ Tests merge_opt_sem function """
+        # Test if overview image changes after inserted optical and sem image
+        md = {
+            model.MD_DESCRIPTION: "green dye",
+            model.MD_BPP: 12,
+            model.MD_BINNING: (1, 1),  # px, px
+            model.MD_PIXEL_SIZE: (1e-6, 1e-6),  # m/px
+            model.MD_POS: (13.7e-3, -30e-3),  # m
+            model.MD_EXP_TIME: 1,  # s
+            model.MD_IN_WL: (600e-9, 620e-9),  # m
+            model.MD_OUT_WL: (620e-9, 650e-9),  # m
+            model.MD_USER_TINT: (0, 0, 255),  # RGB (blue)
+            model.MD_ROTATION: 0.1,  # rad
+            model.MD_SHEAR: 0,
+            model.MD_DIMS: "YXC"
+        }
+
+        opt = model.DataArray(200 * numpy.ones((10, 10, 3)), md)
+        sem = model.DataArray(10 * numpy.ones((10, 10, 3)), md)
+        merged = merge_screen(opt, sem)
+        numpy.testing.assert_equal(True, numpy.any(numpy.not_equal(merged, opt)))
```

Comment:
What does this tests for? It's really hard to follow the logic of this line. It checks that at least one pixel is different? In that case, replacing by just `assert_not_equal(merged, opt)` would make the line clearer.

Reply:
To my knowledge, there doesn't exist an assert_not_equal function for arrays. I changed the code now to 
        self.assertRaises(AssertionError, numpy.testing.assert_array_equal,
                                    merged, opt)

I hope that's easier to read.